src/diffusers/models/embeddings.py

```diff
@@ -371,6 +414,21 @@ def forward(self, text_embeds: torch.Tensor, image_embeds: torch.Tensor):
         embeds = torch.cat(
             [text_embeds, image_embeds], dim=1
         ).contiguous()  # [batch, seq_length + num_frames x height x width, channels]
+
+        if self.use_positional_embeddings:
+            pre_time_compression_frames = (num_frames - 1) * self.temporal_compression_ratio + 1
+            if (
+                self.sample_height != height
```

Comment:
ohh so we are supporting other resolutions now even though the cogvideox original repo says otherwise
do they work once we regenerate positional embeddings here? very nice if it does!

Reply:
It does work and generate good results at some resolutions actually. CogVideoX-5b does not use these embeddings but can generate decent results too due how RoPE is prepared (we don't forbid different resolutions so it can already do this without current PR). CogVideoX-2b, which does use these embeddings, also sees an improvement in quality after these changes. Besides, I think training loras at different resolution could be beneficial, no?

Reply:
ok!